Updates to the 2.2-3.0 migration docs (#14442)
aspnet/AspNetCore.Docs: aspnetcore/migration/22-to-30.md

* Formatting and content negotiation support for `System.Net.HttpClient` - The [Microsoft.AspNet.WebApi.Client](https://www.nuget.org/packages/Microsoft.AspNet.WebApi.Client/) NuGet package provides useful extensibility to `System.Net.HttpClient` with APIs such as `ReadAsAsync`, `PostJsonAsync` etc.

> * Razor Runtime Compilation - Support for runtime compilation of Razor views and pages is now part of [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation)

Reviewer: ```suggestion
* Razor runtime compilation - Support for runtime compilation of Razor views and pages is now part of [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation).
```

> * MVC Newtonsoft.Json Support - Support for using MVC with Newtonsoft.Json is now part of [Microsoft.AspNetCore.Mvc.NewtonsoftJson](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.NewtonsoftJson)

Reviewer: ```suggestion
* MVC `Newtonsoft.Json` support - Support for using MVC with `Newtonsoft.Json` is now part of [Microsoft.AspNetCore.Mvc.NewtonsoftJson](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.NewtonsoftJson).
```

## Json.NET support

As part of the work to [improve the ASP.NET Core shared framework](https://blogs.msdn.microsoft.com/webdev/2018/10/29/a-first-look-at-changes-coming-in-asp-net-core-3-0/), [Json.NET](https://www.newtonsoft.com/json/help/html/Introduction.htm) has been removed from the ASP.NET Core shared framework.

> The default for ASP.NET Core is now [System.Text.Json](https://docs.microsoft.com/en-us/dotnet/api/system.text.json?view=netcore-3.0) which is new in .NET Core 3.0. Consider using `System.Text.Json` when possible because it is high-performance and doesn't require an additional library dependency. However, since `System.Text.Json` is new, it might currently be missing features that your app needs.

Reviewer: ```suggestion
The default for ASP.NET Core is now [System.Text.Json](/dotnet/api/system.text.json?view=netcore-3.0), which is new in .NET Core 3.0. Consider using `System.Text.Json` when possible. It's high-performance and doesn't require an additional library dependency. However, since `System.Text.Json` is new, it might currently be missing features that your app needs.
```

If an app calls `UseMvc` or `UseSignalR`, migrate the app to [Endpoint Routing](xref:fundamentals/routing) if possible. To improve Endpoint Routing compatibility with previous versions of MVC, we've reverted some of the changes in URL generation introduced in ASP.NET Core 2.2. If you experienced problems using Endpoint Routing in 2.2, expect improvements in ASP.NET Core 3.0 with the following exceptions:

* If the app implements `IRouter` or inherits from `Route`, use [DynamicRouteValuesTransformer](https://github.com/aspnet/AspNetCore.Docs/issues/12997) as the replacement.

> * If the app directly accesses `RouteData.Routers` inside MVC to parse URLs you can replace this with use of `LinkParser.ParsePathByEndpointName`. 

Reviewer: ```suggestion
* If the app directly accesses `RouteData.Routers` inside MVC to parse URLs, you can replace this with use of `LinkParser.ParsePathByEndpointName`. 
```

>  - Define the route with a route name

Reviewer: ```suggestion
 * Define the route with a route name.
```

>  - Use `LinkParser.ParsePathByEndpointName` and pass in the desired route name.

Reviewer: ```suggestion
 * Use `LinkParser.ParsePathByEndpointName` and pass in the desired route name.
```
